src/lib/tcti_ldr.c

```diff
@@ -168,49 +168,40 @@ tcti_conf tcti_get_config(void) {
     };
 
     /* no tcti config supplied, get it from env */
-    char *optstr = getenv (TPM2_PKCS11_TCTI);
-    if (!optstr) {
+    char *env_conf_string = getenv(TPM2_PKCS11_TCTI);
```

Comment:
Their is a bug, but your patch doesn't do the bare minimum to fix that bug. Ie 1 patch/commit to fix the bug and another if you want to do refactoring.

I don't see optstr and env_conf_string needing to exist.... you can just "reseat the pointer on optstr" when you need to do the duplication below.

Reply:
Thanks for your feedback. I'm definitely not as nimble in C as I am in other programming languages, and I appreciate you taking the time to help me along. I'll try to explain the thought process that went into this patch a little further, so that any flaws in my reasoning become more apparent.

1. Your point about not including unnecessary refactoring in a bug fix is well taken. I didn't introduce the refactoring on a whim, but was trying to address two issues:
    - Branching: I indeed expected to have to `free()` the `conf_string` before the method return (more on that later). Having to do that would have introduced a fairly inelegant label with a number of corresponding `goto` statements, so, instead of doing that, I decided to refactor the branches.
    - Variable naming: The method is trying to populate a `tcti_conf` string with `name` and `opts` fields. Since the `optstr` variable is expected to contain information about both fields, it felt like `confstr` would be a more descriptive name. I then went from `confstr` to `conf_string` based on the coding standards document included in this project (multiple words separated by underscore, no abbreviations except for some exceptions). Rigorously applying the coding standard would have led to `configuration_string` and `environment_configuration_string`, but this would have been inconsistent with the rest of the project (`tcti_conf`, `getenv()` etc.), so I decided against it.
2. `optstr` and `env_conf_string` indeed don't need to coexist, and my patch introduced one as a replacement for the other. As also touched upon above, the renaming came about because I was unsure what name I would use for the copied `optstr`. Both `tmp_optstr` and `optstr_copy` felt a little yuck.
3. I decided on a blanket copy of the environment variable because of [uncertainty](https://stackoverflow.com/questions/19745648/when-getenv-can-alter-content-of-previously-returned-pointer-when-using-glibc) about how subsequent calls to `setenv()` or `putenv()` could potentially affect the value that the pointer obtained from `getenv()` is pointing to.